build.sbt

```diff
@@ -7,7 +7,8 @@ def scala3 = "3.1.0"
 def scala2Versions = List(scala212, scala211, scala213)
 def allScalaVersions = scala2Versions :+ scala3
 
-def scalajs = "1.7.1"
+def scala2js = "1.8.0"
+def scala3js = "1.7.1"
```

Comment:
There's no reason to make those different. You can always use 1.8.0, since Scala.js IR is backward binary compatible.

Reply:
Right, except we seem to be running into forward-compatibility problem here only on Scala 3.
```
mdoc.internal.markdown.ModifierException: mdoc:js exception
Caused by: org.scalajs.ir.IRVersionNotSupportedException: Failed to deserialize a file compiled with Scala.js 1.8 (supported up to: 1.7): /home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-js/scalajs-library_2.13/1.8.0/scalajs-library_2.13-1.8.0.jar:/scala/volatile.sjsir
```

Reply:
That must be due to something in your infrastructure that I am not familiar with. This is because the *linker* being used is 1.7.x instead of being 1.8.0. But Scala 3 per se doesn't even get to *choose* that. It must be something else in your infrastructure that selects the 1.7.x linker, whereas it should always select the 1.8.0 linker.

Reply:
Thanks. So @tgodzik it's back to my original question, where else do I need to change the version number 😆

Reply:
No idea :sweat: There seem to be no place where we define `1.7` linker so that must be brought in by the compiler maybe? Doesn't the Scala 3 compiler need to be able to read 1.8 ?

Reply:
For locally released version everything seems to be correct:
```
cs resolve -t org.scalameta:mdoc-js_3:2.2.22+71-5ab2f66a+20211222-1821-SNAPSHOT | grep -B 12 -A 10 linker                           
  Result:
└─ org.scalameta:mdoc-js_3:2.2.22+71-5ab2f66a+20211222-1821-SNAPSHOT
   ├─ org.scala-js:scalajs-linker_2.13:1.8.0
   │  ├─ com.google.javascript:closure-compiler:v20211201
   │  ├─ org.scala-js:scalajs-ir_2.13:1.8.0
   │  │  └─ org.scala-lang:scala-library:2.13.6 -> 2.13.7
   │  ├─ org.scala-js:scalajs-linker-interface_2.13:1.8.0
   │  │  ├─ org.scala-js:scalajs-ir_2.13:1.8.0
   │  │  │  └─ org.scala-lang:scala-library:2.13.6 -> 2.13.7
   │  │  ├─ org.scala-js:scalajs-logging_2.13:1.1.1
   │  │  │  └─ org.scala-lang:scala-library:2.13.2 -> 2.13.7
   │  │  └─ org.scala-lang:scala-library:2.13.6 -> 2.13.7
   │  ├─ org.scala-lang:scala-library:2.13.6 -> 2.13.7
   │  └─ org.scala-lang.modules:scala-parallel-collections_2.13:0.2.0
   │     └─ org.scala-lang:scala-library:2.13.0 -> 2.13.7
   ├─ org.scala-lang:scala3-library_3:3.1.0
   │  └─ org.scala-lang:scala-library:2.13.6 -> 2.13.7
```

Reply:
@sjrd what would cause it to always work after you upgrade version such as https://github.com/lampepfl/dotty/pull/13734/files ?

Reply:
IIUC (which I very well may not) the Scala 3 compiler uses the `.class` files of dependencies for compiling. Linking is what uses the `.sjsir` files and depends on the IR version (hence it is the linker throwing this error), but there is no Scala 3 linker—as described in this [blog post](https://www.scala-lang.org/2020/11/03/scalajs-for-scala-3.html):
> The .sjsir format and specification is independent of Scala, its compiler or its standard library. In fact, there is nothing Scala-specific in the entire linker, other than a few optimizations targeted at Scala-like code. This is important because it means that the linker is independent of the version of Scala: we use the same linker irrespective of the version of Scala that was used to create the .sjsir files.

Reply:
Classpath for the compiler seems to be:
```
/home/tgodzik/Documents/mdoc/tests/jsdocs/target/scala-3.1.0/classes:/home/tgodzik/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala3-library_sjs1_3/3.1.0/scala3-library_sjs1_3-3.1.0.jar:/home/tgodzik/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-js/scalajs-library_2.13/1.8.0/scalajs-library_2.13-1.8.0.jar:/home/tgodzik/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-js/scalajs-dom_sjs1_3/2.0.0/scalajs-dom_sjs1_3-2.0.0.jar:/home/tgodzik/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.13.6/scala-library-2.13.6.jar
```
And the compiler isn't able to read? Should the classpath contain the linker?